app/code/Magento/Catalog/Console/Command/ImagesResizeCommand.php

```diff
@@ -62,39 +67,86 @@ protected function execute(
         \Symfony\Component\Console\Input\InputInterface $input,
         \Symfony\Component\Console\Output\OutputInterface $output
     ) {
+        $options = $input->getOptions();
         $this->appState->setAreaCode(\Magento\Framework\App\Area::AREA_GLOBAL);
 
         /** @var \Magento\Catalog\Model\ResourceModel\Product\Collection $productCollection */
         $productCollection = $this->productCollectionFactory->create();
         $productIds = $productCollection->getAllIds();
+        $productCount = count($productIds);
+        $limit = $this->getLimit($options);
+        $offset = $this->getOffset($options);
+
+        if ($limit) {
+            $productIds = array_slice($productIds, $offset, $limit);
+        }
+
+        $maxOffset = $productCount - 1;
+        if ($offset && $offset > $maxOffset) {
+            $output->writeln("<error>Offset may not be higher than $maxOffset</error>");
+            return Cli::RETURN_FAILURE;
+        }
+
         if (!count($productIds)) {
             $output->writeln("<info>No product images to resize</info>");
             // we must have an exit code higher than zero to indicate something was wrong
-            return \Magento\Framework\Console\Cli::RETURN_SUCCESS;
+            return Cli::RETURN_SUCCESS;
         }
 
+        $processes = $this->getProcessesAmount($options);
+        /** @var \Magento\Catalog\Model\Product\Image\Process\Queue $queue */
+        $queue = $this->queueFactory->create(
+            [
+                'maxProcesses' => $processes,
+                'output' => $output,
+            ]
+        );
+
         try {
-            foreach ($productIds as $productId) {
-                try {
-                    /** @var \Magento\Catalog\Model\Product $product */
-                    $product = $this->productRepository->getById($productId);
-                } catch (\Magento\Framework\Exception\NoSuchEntityException $e) {
-                    continue;
-                }
-
-                /** @var \Magento\Catalog\Model\Product\Image\Cache $imageCache */
-                $imageCache = $this->imageCacheFactory->create();
-                $imageCache->generate($product);
-
-                $output->write(".");
-            }
+            $queue->setProducts($productIds);
+            $output->writeln("<info>Resizing product images</info>");
+            $queue->process();
         } catch (\Exception $e) {
             $output->writeln("<error>{$e->getMessage()}</error>");
             // we must have an exit code higher than zero to indicate something was wrong
-            return \Magento\Framework\Console\Cli::RETURN_FAILURE;
+            return Cli::RETURN_FAILURE;
         }
 
         $output->write("\n");
         $output->writeln("<info>Product images resized successfully</info>");
+        return Cli::RETURN_SUCCESS;
+    }
+
+    /**
+     * @param array $options
```

Comment:
It would be nice to specify information about the variables in the docblock for each method.